core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -324,6 +324,21 @@ public String getUrl() {
         }
 
         /**
+         * URL of specific updateSite.
+         */
+        public String getUrl(UpdateSite site) {
```

Comment:
well, this is the URL for the update Site and for a specific download service, I can move it to UpdateSite and send the DownloadService id I want to retrieve the URL for, or keep it here and pass the UpdateSite as parameter..I will go with your suggestion although they are inter-dependent and I do not see a big difference.